static/src/javascripts/projects/common/modules/atoms/youtube.js

```diff
@@ -275,21 +277,30 @@ const shouldAutoplay = (atomId: string): boolean => {
         config.get('page.contentType', '').toLowerCase() === 'video';
 
     const isFront = () => config.get('page.isFront');
+    const isUSContent =
+        config.get('page.productionOffice', '').toLowerCase() === 'us';
 
-    if (isUSContent) {
-        const isPaidContent = config.get('page.isPaidContent');
-        return (
-            ((isVideoArticle() && isMainVideo()) || isFront()) &&
-            isPaidContent &&
-            flashingElementsAllowed()
-        );
+    const isPaidContent = config.get('page.isPaidContent');
+    const isUsPaidContentVideo =
+        isUSContent &&
+        isPaidContent &&
+        ((isVideoArticle() && isMainVideo()) || isFront()) &&
+        flashingElementsAllowed();
+    if (isUsPaidContentVideo) {
+        return {
+            autoplay: isUsPaidContentVideo,
+            mutedOnStart: isUsPaidContentVideo,
+        };
     }
-    return (
-        ((isVideoArticle() && isInternalReferrer() && isMainVideo()) ||
-            isFront()) &&
-        !isAutoplayBlockingPlatform() &&
-        flashingElementsAllowed()
-    );
+
+    return {
+        autoplay:
+            ((isVideoArticle() && isInternalReferrer() && isMainVideo()) ||
```

Comment:
Be great to see some tests for this logic, because it is pretty hard to parse as a human! I realise this PR is just a change and therefore there may already be some tests, but if this change didn't entail updating a test then at the very least I suspect we need to add a new test case?

Doesn't have to block this release if you're in a hurry, but I'm always more comfortable when these kind of logic knots are well tested!

Reply:
We had the same conversation. We need to rush this out but will raise a card on Trello to add unit tests. This entire module is missing unit testing for most of its logic so it would require a good chunk of work, but it's something I'd want to sort out myself too.